internal/filesystem/ipfs/files.go

```diff
@@ -0,0 +1,454 @@
+package ipfs
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"io"
+	"io/fs"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/djdv/go-filesystem-utils/internal/filesystem"
+	fserrors "github.com/djdv/go-filesystem-utils/internal/filesystem/errors"
+	"github.com/djdv/go-filesystem-utils/internal/generic"
+	"github.com/ipfs/boxo/mfs"
+	shell "github.com/ipfs/go-ipfs-api"
+	"github.com/multiformats/go-multiaddr"
+	manet "github.com/multiformats/go-multiaddr/net"
+)
+
+type (
+	FilesFS struct {
+		ctx    context.Context
+		cancel context.CancelFunc
+		shell  *shell.Shell
+		info   nodeInfo
+	}
+	FilesOption  func(*FilesFS) error
+	filesOptions []FilesOption
+	filesShared  struct {
+		ctx         context.Context
+		cancel      context.CancelFunc
+		shell       *shell.Shell
+		mountTime   *time.Time // Substitute for mtime (no data in UFSv1).
```

Comment:
This is going to confuse some programs when they modify the file and the timestamp does not change.
We should instead have a unique time value for open files, which gets updated on the usual events (like `write`, etc.).

Using `time.Now` isn't usually a good option for the opposite reason, programs will think a file has always been modified. This causes some tight reload loops in various programs that monitor for this.